"use strict";(self.webpackChunkdocs=self.webpackChunkdocs||[]).push([[8246],{9869:(e,o,n)=>{n.r(o),n.d(o,{assets:()=>a,contentTitle:()=>i,default:()=>h,frontMatter:()=>l,metadata:()=>s,toc:()=>d});var r=n(4848),t=n(8453);const l={sidebar_position:4},i="Troubleshooting",s={id:"rad-lab-launcher/troubleshooting",title:"Troubleshooting",description:"Overview",source:"@site/docs/rad-lab-launcher/troubleshooting.md",sourceDirName:"rad-lab-launcher",slug:"/rad-lab-launcher/troubleshooting",permalink:"/rad-lab/docs/rad-lab-launcher/troubleshooting",draft:!1,unlisted:!1,tags:[],version:"current",sidebarPosition:4,frontMatter:{sidebar_position:4},sidebar:"tutorialSidebar",previous:{title:"Deployment via RAD Lab Launcher",permalink:"/rad-lab/docs/rad-lab-launcher/launcher_deployment/launcher"},next:{title:"RAD Lab UI",permalink:"/rad-lab/docs/category/rad-lab-ui"}},a={},d=[{value:"Overview",id:"overview",level:2},{value:"Deployment Troubleshooting",id:"deployment-troubleshooting",level:2},{value:"Google Organization Policies - unable to modify constraints",id:"google-organization-policies---unable-to-modify-constraints",level:3},{value:"Project Quota Exceeded",id:"project-quota-exceeded",level:3},{value:"Timeout when Destroying the deployment",id:"timeout-when-destroying-the-deployment",level:3},{value:"Operations Troubleshooting",id:"operations-troubleshooting",level:2},{value:"Local Terraform Deployment ID Directory Already Exists",id:"local-terraform-deployment-id-directory-already-exists",level:3}];function c(e){const o={a:"a",code:"code",h1:"h1",h2:"h2",h3:"h3",li:"li",ol:"ol",p:"p",pre:"pre",strong:"strong",ul:"ul",...(0,t.R)(),...e.components};return(0,r.jsxs)(r.Fragment,{children:[(0,r.jsx)(o.h1,{id:"troubleshooting",children:"Troubleshooting"}),"\n",(0,r.jsx)(o.h2,{id:"overview",children:"Overview"}),"\n",(0,r.jsx)(o.p,{children:"The troubleshooting section aims to identify the most common problems users may face when deploying and using RAD Lab."}),"\n",(0,r.jsxs)(o.p,{children:["If you can't find a solution, please don't hesitate to create a ",(0,r.jsx)(o.a,{href:"https://github.com/GoogleCloudPlatform/rad-lab/issues",children:"GitHub issue"}),"."]}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"NOTE:"}),"  This is not an officially supported Google product"]}),"\n",(0,r.jsxs)(o.ol,{children:["\n",(0,r.jsx)(o.li,{children:(0,r.jsx)(o.a,{href:"/rad-lab/docs/rad-lab-launcher/troubleshooting#project-quota-exceeded",children:"Project quota exceeded"})}),"\n",(0,r.jsx)(o.li,{children:(0,r.jsx)(o.a,{href:"/rad-lab/docs/rad-lab-launcher/troubleshooting#google-organization-policies---unable-to-modify-constraints",children:"Unable to modify Organization Policy Constraints"})}),"\n",(0,r.jsx)(o.li,{children:(0,r.jsx)(o.a,{href:"/rad-lab/docs/rad-lab-launcher/troubleshooting#local-terraform-deployment-id-directory-already-exists",children:"Local Terraform Deployment ID Directory Already Exists"})}),"\n",(0,r.jsx)(o.li,{children:(0,r.jsx)(o.a,{href:"/rad-lab/docs/rad-lab-launcher/troubleshooting#timeout-when-destroying-the-deployment",children:"Timeout when Destroying the deployment"})}),"\n"]}),"\n",(0,r.jsx)(o.h2,{id:"deployment-troubleshooting",children:"Deployment Troubleshooting"}),"\n",(0,r.jsx)(o.h3,{id:"google-organization-policies---unable-to-modify-constraints",children:"Google Organization Policies - unable to modify constraints"}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"Issue"}),": When running the ",(0,r.jsx)(o.code,{children:"python3 radlab.py"})," installation from Cloud Shell, you receive the following error:"]}),"\n",(0,r.jsx)(o.pre,{children:(0,r.jsx)(o.code,{children:"Error: Error waiting to create Instance: Error waiting for Creating Instance: Error code 25, message: Constraint constraints/compute.vmExternalIpAccess violated for project [project_id]. Add instance projects/[project_id]/zones/us-east4-c/instances/notebooks-instance-0 to the constraint to use external IP with it.\n.\n.\nError Occurred - Deployment failed for ID:[deployment_id]\n"})}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"Solution"}),": If you see above error in your initial deployment run, rerun the deployment via ",(0,r.jsx)(o.code,{children:"python3 radlab.py"})," using the [deployment_id] and select ",(0,r.jsx)(o.code,{children:"Update"})," (in ",(0,r.jsx)(o.a,{href:"/rad-lab/docs/rad-lab-launcher/launcher_deployment/launcher#deploy-a-rad-lab-module",children:"Steps to Deploy RAD Lab Modules"}),"). This may have been caused as the Organization Policy ",(0,r.jsx)(o.code,{children:"constraints/compute.vmExternalIpAccess"})," is not completely rolled out."]}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"NOTE:"})," Similarly, if the error occurs for any other organization policies, then the workaround is same as above."]}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"Issue"}),":\nWhen running the ",(0,r.jsx)(o.code,{children:"python3 radlab.py"})," installation from Cloud Shell, you receive the following error(s):"]}),"\n",(0,r.jsx)(o.pre,{children:(0,r.jsx)(o.code,{children:"Error: googleapi: Error 403: The caller does not have permission, forbidden\n...\nwith google_project_organization_policy.shielded_vm_policy[0],\n...\n"})}),"\n",(0,r.jsx)(o.p,{children:"OR"}),"\n",(0,r.jsx)(o.pre,{children:(0,r.jsx)(o.code,{children:"with google_project_organization_policy.trustedimage_project_policy[0], ...\n"})}),"\n",(0,r.jsx)(o.p,{children:"OR"}),"\n",(0,r.jsx)(o.pre,{children:(0,r.jsx)(o.code,{children:"with google_project_organization_policy.external_ip_policy[0],\n"})}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"Solution"}),": The project is required to be run as part of a Google Organization in Google Cloud. Be sure that the Google Cloud user you run the ",(0,r.jsx)(o.code,{children:"python3 radlab.py"})," script as has both of the following roles:"]}),"\n",(0,r.jsxs)(o.ul,{children:["\n",(0,r.jsx)(o.li,{children:"Organization Policy Administrator"}),"\n",(0,r.jsx)(o.li,{children:"Organization Viewer"}),"\n"]}),"\n",(0,r.jsx)(o.h3,{id:"project-quota-exceeded",children:"Project Quota Exceeded"}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"Issue"}),": When running the ",(0,r.jsx)(o.code,{children:"python3 radlab.py"})," installation from Cloud Shell, you receive the following error:"]}),"\n",(0,r.jsx)(o.pre,{children:(0,r.jsx)(o.code,{children:' Error: Error setting billing account "[yourBillingID]" for project "projects/radlab-ds-analytics-[deployment_id]": \n googleapi: Error 400: Precondition check failed., \n failedPrecondition\n\n  with module.project_radlab_ds_analytics.module.project-factory.google_project.main,\n'})}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"Solution"}),": There are soft limits to the number of projects you can initially associate with  your billing account. You can request a quota increase for the number of projects you are allowed to link to your billing account by filling out the form ",(0,r.jsx)(o.a,{href:"https://support.google.com/code/contact/billing_quota_increase",children:"here."}),"\nPlease be sure you are logged in as a user with Google Cloud project Owner rights."]}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"Note:"})," If your billing account is relatively new, or if you are still in your free trial period,  you may be required to authorize funds to pre-pay your billing account. The amount you asked to pay will vary depending on your billing history but will usually not exceed $50. Please see ",(0,r.jsx)(o.a,{href:"https://support.google.com/cloud/answer/6330231?hl=en#",children:"Why am I being asked to make a payment for more projects"})," for details."]}),"\n",(0,r.jsx)(o.h3,{id:"timeout-when-destroying-the-deployment",children:"Timeout when Destroying the deployment"}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"Issue"}),": When running the ",(0,r.jsx)(o.code,{children:"python3 radlab.py"})," installation from Cloud Shell, you receive the following error:"]}),"\n",(0,r.jsx)(o.pre,{children:(0,r.jsx)(o.code,{children:"\u2577\n\u2502 Error: Error waiting for Deleting Subnetwork: timeout while waiting for state to become 'DONE' (last state: 'RUNNING', timeout: 6m0s)\n\u2502 \n.\n.\nError Occurred - Deployment failed for ID:[deployment_id]\n"})}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"Solution"}),": If you see above error, rerun the deployment via ",(0,r.jsx)(o.code,{children:"python3 radlab.py"})," using the [deployment_id] and select ",(0,r.jsx)(o.code,{children:"Delete"})," (in ",(0,r.jsx)(o.a,{href:"/rad-lab/docs/rad-lab-launcher/launcher_deployment/launcher#deploy-a-rad-lab-module",children:"Steps to Deploy RAD Lab Modules"}),"). This may have been caused if it took longer than expected to destroy any resource(s)."]}),"\n",(0,r.jsx)(o.h2,{id:"operations-troubleshooting",children:"Operations Troubleshooting"}),"\n",(0,r.jsx)(o.h3,{id:"local-terraform-deployment-id-directory-already-exists",children:"Local Terraform Deployment ID Directory Already Exists"}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"Issue"}),":  When running an ",(0,r.jsx)(o.code,{children:"Update"})," or ",(0,r.jsx)(o.code,{children:"Delete"})," action when running ",(0,r.jsx)(o.code,{children:"python3 radlab.py"})," installation from Cloud Shell, you receive the following error:"]}),"\n",(0,r.jsx)(o.pre,{children:(0,r.jsx)(o.code,{children:"File \"radlab.py\", line 134, in main\n\n    os.mkdir(env_path)\n\nFileExistsError: [Errno 17] File exists:\n '/home/[radlabAdminUser]/radlab/deployments/data_science_[deployment_id]'\n"})}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"Solution"}),":  You likely have a local copy of a Terraform ",(0,r.jsx)(o.code,{children:"/deployments"})," folder with the same deploymentID in your Cloud Shell instance that needs to be removed. This can happen if you  perform multiple Actions on a single deployment from the same Cloud Shell instance. Even though you are looking to a shared Google Cloud bucket to get the current Terraform state, Terraform will still try to create a local copy of the deployment first."]}),"\n",(0,r.jsxs)(o.p,{children:["To solve this issue, you can safely remove the conflicting deployment folder by navigating to the ",(0,r.jsx)(o.code,{children:"./radlab-launcher/deployments"})," directory in Cloud Shell"]}),"\n",(0,r.jsx)(o.pre,{children:(0,r.jsx)(o.code,{className:"language-bash",children:"cd ./deployments\n"})}),"\n",(0,r.jsx)(o.p,{children:"And removing the module folder with the conflicting deploymentID:"}),"\n",(0,r.jsx)(o.pre,{children:(0,r.jsx)(o.code,{className:"language-bash",children:"rm -rfv /data_science_[deployment_id]/\n"})}),"\n",(0,r.jsxs)(o.p,{children:[(0,r.jsx)(o.strong,{children:"Note:"})," The above command will remove all files and sub-directories in the ",(0,r.jsx)(o.code,{children:"/data_science_[deployment_id]/"})," directory before removing the ",(0,r.jsx)(o.code,{children:"/data_science_[deployment_id]/"})," directory. The contents of this directory will sync from the Google Cloud state bucket next time you run the ",(0,r.jsx)(o.code,{children:"python3 radlab.py"})," installation."]})]})}function h(e={}){const{wrapper:o}={...(0,t.R)(),...e.components};return o?(0,r.jsx)(o,{...e,children:(0,r.jsx)(c,{...e})}):c(e)}},8453:(e,o,n)=>{n.d(o,{R:()=>i,x:()=>s});var r=n(6540);const t={},l=r.createContext(t);function i(e){const o=r.useContext(l);return r.useMemo((function(){return"function"==typeof e?e(o):{...o,...e}}),[o,e])}function s(e){let o;return o=e.disableParentContext?"function"==typeof e.components?e.components(t):e.components||t:i(e.components),r.createElement(l.Provider,{value:o},e.children)}}}]);